### PR TITLE
Implement Adamax Optimizer Method for CNTK

### PR DESCRIPTION
**Pull request by yuanhaitao**

### CODE REVIEW

File: Source/Math/CPUMatrix.cpp [hunk 1]
```diff
@@ -1293,6 +1293,54 @@ void CPUMatrix<ElemType>::Adam(CPUMatrix<ElemType>& gradients, CPUMatrix<ElemTyp
...
+		ElemType ada = adaWeight * smoothAda[i];
+		if (ada <abs(g))
+		{
+			ada = abs(g);
+		}
+		
+
+		smoothAda[i] = ada;
+		//		ElemType ada = sqrt(adaSqr);
+		ElemType w = adaMul * (ElemType)(1.0 / (ada + 1e-8));
```
Comment (KeDengMS) [inline Source/Math/CPUMatrix.cpp]: Can you make epsilon a variable instead of fixed 1e-8?

**Comment by yuanhaitao** [comment]
Thanks for your comment, I’ll try to do it

**Comment by KeDengMS** [comment]
Just noticed another [PR ](https://github.com/Microsoft/CNTK/pull/1812) to add Adamax with an option in existing Adam learner. I believe that's a better way to reuse code. Will let you know once it's merged.

**Comment by KeDengMS** [comment]
Adamax is available now as an option to Adam. https://github.com/Microsoft/CNTK/pull/1812

**Pull request closed without merging**